From e6b22cd6433618891178ced990d646a097ef15eb Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa@datto.com>
Date: Fri, 30 Mar 2018 17:01:14 -0400
Subject: [PATCH 7/9] Remove default values for keypair and security names for
 EC2LatentBuildSlave

This is a simpler/trivial adaptation of 68a9267d5fff06e0ff7c6ea8a82ab66fcf6a359c in buildbot 0.9.x,
originally authored by Vladimir Rutsky <vladimir@rutsky.org>.

There are no test adaptations for this, as ultimate behavior has not yet changed.

Signed-off-by: Neal Gompa <ngompa@datto.com>
---
 master/buildbot/buildslave/ec2.py | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/master/buildbot/buildslave/ec2.py b/master/buildbot/buildslave/ec2.py
index 00ead8a5d..622bcc973 100644
--- a/master/buildbot/buildslave/ec2.py
+++ b/master/buildbot/buildslave/ec2.py
@@ -55,8 +55,8 @@ class EC2LatentBuildSlave(AbstractLatentBuildSlave):
                  valid_ami_owners=None, valid_ami_location_regex=None,
                  elastic_ip=None, identifier=None, secret_identifier=None,
                  aws_id_file_path=None, user_data=None, region=None,
-                 keypair_name='latent_buildbot_slave',
-                 security_name='latent_buildbot_slave',
+                 keypair_name=None,
+                 security_name=None,
                  max_builds=None, notify_on_missing=[], missing_timeout=60 * 20,
                  build_wait_timeout=60 * 10, properties={}, locks=None,
                  spot_instance=False, max_spot_price=1.6, volumes=[],
@@ -89,6 +89,12 @@ class EC2LatentBuildSlave(AbstractLatentBuildSlave):
             else:
                 # verify that regex will compile
                 re.compile(valid_ami_location_regex)
+        if keypair_name is None:
+            keypair_name = 'latent_buildbot_slave'
+            log.msg('Using default keypair name, since none is set')
+        if security_name is None:
+            security_name = 'latent_buildbot_slave'
+            log.msg('Using default keypair name, since none is set')
         self.valid_ami_owners = valid_ami_owners
         self.valid_ami_location_regex = valid_ami_location_regex
         self.instance_type = instance_type
-- 
2.14.3

